python/mxnet/io.py

```diff
@@ -141,12 +141,23 @@ def __str__(self):
             label_shapes)
 
 class DataIter(object):
-    """The base class of a data iterator.
+    """The base class for an MXNet data iterator.
+
+    All I/O in MXNet is handled by specializations of this class. Data iterators
+    in MXNet are similar to standard-iterators in Python. On each call to `next`
+    they return a `DataBatch` which represents the next batch of data. When
+    there is no more data to return, it raises a `StopIteration` exception.
 
     Parameters
     ----------
     batch_size : int, optional
-        The batch size, namely the number of examples in a batch.
+        The batch size, namely the number of items in the batch.
+
+    See Also
+    --------
+    NDArrayIter : Data-iterator for MXNet NDArray or numpy-ndarray objects.
+    CSVIter : Data-iterator for csv data.
+    ImageIter : Data-iterator for images.
```

Comment:
@piiswrong Isn't ImageIter a subclass of this DataIter - [reference](https://github.com/dmlc/mxnet/blob/f659ef599fd3e332a21bd62653a2795b61eea8b7/python/mxnet/image.py#L446). 

ImageRecordIter seems to be coming from the [c++ core](https://github.com/dmlc/mxnet/blob/8edd97c4a0349ec8ffeb10e5abd96e194350fd31/src/io/iter_image_recordio_2.cc#L495) which is extending dmlc's DataIter (ref: [this](https://github.com/dmlc/mxnet/blob/8edd97c4a0349ec8ffeb10e5abd96e194350fd31/src/io/iter_image_recordio_2.cc#L559) and [this](https://github.com/dmlc/mxnet/blob/e7514fe1b3265aaf15870b124bb6ed0edd82fa76/include/mxnet/io.h#L24))

Unless I'm missing something which can't be ruled out completely :)